src/connectors/MagicConnector.ts

```diff
@@ -0,0 +1,111 @@
+import { ChainId, ProviderType } from '@dcl/schemas'
+import { ConnectorUpdate } from '@web3-react/types'
+import { getConfiguration } from '../configuration'
+import { AbstractConnector } from './AbstractConnector'
+// tslint:disable-next-line
+import type { InstanceWithExtensions, SDKBase } from '@magic-sdk/provider'
+// tslint:disable-next-line
+import type { OAuthExtension } from '@magic-ext/oauth'
+
+export class MagicConnector extends AbstractConnector {
+  private chainId: ChainId
+  private account: string | null
+  private magic: InstanceWithExtensions<SDKBase, OAuthExtension[]> | undefined
+
+  constructor(desiredChainId: ChainId) {
+    super({ supportedChainIds: getConfiguration()[ProviderType.MAGIC].chains })
+    this.chainId = desiredChainId
+    this.account = null
+  }
+
+  activate = async (): Promise<ConnectorUpdate<string | number>> => {
+    this.magic = await this.buildMagicInstance(this.chainId)
+    const isLoggedIn = await this.magic.user.isLoggedIn()
+    if (!isLoggedIn) {
+      return {
+        provider: null,
+        account: null,
+        chainId: this.chainId
+      }
+    }
+    const provider = await this.getProvider()
+    const accounts: string[] = (await provider.request({ method: 'eth_accounts' }))
+    this.account = accounts[0] ?? null
+
+    return {
+      provider,
+      chainId: this.chainId,
+      account: this.account
+    }
+  }
+
+  getProvider = async (): Promise<any> => {
+    if (!this.magic) {
+      throw new Error('Magic: instance was not initialized')
+    }
+
+    const provider = await this.magic.wallet.getProvider()
+    return new Proxy(provider.request, {
```

Comment:
Where does this Proxy thing come from? Doesn't seem new in the PR nor it is imported anywhere. Is it a JS thing?

Reply:
Proxy [is part of the ECMAS language specification.](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Proxy)